File: PRACTICA1/RUBY/Deepspace/PowerEfficientSpaceStation.rb
#encoding:utf-8
require './PowerEfficientSpaceStationToUI'
module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class PowerEfficientSpaceStation < SpaceStation
    @EFFICIENCYFACTOR = 1.1 #float  
    def initialize(estacion)  #SpaceStation
        copy(estacion)   
    end
    
    def setLoot(loot) 
        super 
        if loot.efficient 
            return Transformation::GETEFFICIENT
        else
            return Transformation::NOTRANSFORM
        end
    end

    
    def fire    #devuelve un float
        return PowerEfficientSpaceStation.efficiencyFactor*super
    end
    
    def protection 
        return PowerEfficientSpaceStation.efficiencyFactor*super
    end

    def self.efficiencyFactor 
        @EFFICIENCYFACTOR
    end

    def getUIversion
        PowerEfficientSpaceStationToUI.new(self)
      end

    def to_s
        out = "Estación espacial eficiente: "
        out += super
    end

end 
end

File: PRACTICA1/JAVA/deepspace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Celia Arias Martínez, Lucía Salamanca López
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    
    private static final float EFFICIENCYFACTOR=1.10f; 
    
    public PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        Transformation trans = super.setLoot(loot);
        if (trans==Transformation.SPACECITY)
            return Transformation.NOTRANSFORM;
        else 
            return trans;      
    }
    
    @Override
    public PowerEfficientSpaceStationToUI getUIversion(){
         return new PowerEfficientSpaceStationToUI(this);
    }
    
    
}
